Public/Src/Cache/ContentStore/Distributed/NuCache/RaidedRedisDatabase.cs

```diff
@@ -84,10 +87,12 @@ public async Task<BoolResult> ExecuteRedisAsync(OperationContext context, Func<R
         }
 
         /// <nodoc />
-        public async Task<(TResult primary, TResult secondary)> ExecuteRaidedAsync<TResult>(OperationContext context, Func<RedisDatabaseAdapter, Task<TResult>> executeAsync, bool concurrent = true, [CallerMemberName]string caller = null)
+        public async Task<(TResult primary, TResult secondary)> ExecuteRaidedAsync<TResult>(OperationContext context, Func<RedisDatabaseAdapter, CancellationToken, Task<TResult>> executeAsync, TimeSpan? retryWindow, bool concurrent = true, [CallerMemberName]string caller = null)
             where TResult : BoolResult
         {
-            var primaryResultTask = ExecuteAndCaptureRedisErrorsAsync(PrimaryRedisDb, executeAsync);
+            using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(context.Token);

```

Comment:
This using needs a {} block.

Reply:
Based on https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement, I thought it'd be fine to do "using var" since calls Dispose() on the token source at the end of the method, unless we'd like to restrict to only a portion of this method.

Reply:
This is a new feature in C# 8:)

---
In reply to: [348761467](https://github.com/microsoft/BuildXL/pull/1233#discussion_r348761467) [](ancestors = 348761467)